GitUI/FindFilePredicateProvider.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using GitCommands;
+
+namespace GitUI
+{
+    public interface IFindFilePredicateProvider
+    {
+        Func<string, bool> Get(string name, string workingDir);
+    }
+
+    public sealed class FindFilePredicateProvider : IFindFilePredicateProvider
+    {
+        public Func<string, bool> Get(string name, string workingDir)
+        {
+            var pattern = name.ToPosixPath();
+            var dir = workingDir.ToPosixPath();
```

Comment:
both input parameters need to be validated
throw `ArgumentException` if either null or whitespace